src/tools/rbd/action/Import.cc

```diff
@@ -132,6 +138,30 @@ static int do_import(librbd::RBD &rbd, librados::IoCtx& io_ctx,
 #endif
   }
 
+  if (import_snaps || !from_stdin) {
+    r = safe_read_exact(fd, image_buf, utils::RBD_IMAGE_BANNER.size());
+    if (r < 0)
+      goto done;
+    image_buf[utils::RBD_IMAGE_BANNER.size()] = '\0';
+    if (strcmp(image_buf, utils::RBD_IMAGE_BANNER.c_str())) {
+      // Old format
+      if (from_stdin) {
+	// --with-snaps specified for an old format exported file
+	r = -EINVAL;
+	goto done;
+      }
+      // seek back to the beginning.
+      lseek(fd, 0, SEEK_SET);
+    } else {
+      r = safe_read_exact(fd, &size, 8);
```

Comment:
I'd hate to automatically switch to the new format based on the header -- I'd rather see a warning message displayed saying something along the lines of "image appears to contain snapshots, import with --with-snaps".